### PR TITLE
geant4 10.6.2

### PR DESCRIPTION
**Pull request by cxwx**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/geant4.rb [hunk 1]
```diff
@@ -14,70 +14,19 @@ class Geant4 < Formula
...
-    url "https://cern.ch/geant4-data/datasets/G4RealSurface.2.1.1.tar.gz"
-    sha256 "90481ff97a7c3fa792b7a2a21c9ed80a40e6be386e581a39950c844b2dd06f50"
-  end
-
   def install
     mkdir "geant-build" do
       args = std_cmake_args + %w[
         ../
         -DGEANT4_USE_GDML=ON
-        -DGEANT4_BUILD_MULTITHREADED=ON
```
Comment (SMillerDev) [inline Formula/geant4.rb:80]: why not?

Comment (cxwx) [reply Formula/geant4.rb:80]: multithreaded option conflict with python option on Mac OS X. so either MULTITHREADED or PYTHON is OK, but not both

Comment (cxwx) [reply Formula/geant4.rb:80]: I think the best way is split geant4 formula into to formula name geant4 (with python) and geant4MT without python and set them conflict with each other

Comment (cxwx) [reply Formula/geant4.rb:80]: detail is in `https://github.com/Homebrew/homebrew-core/pull/58642/checks?check_run_id=912013459`

Comment (SeekingMeaning) [reply Formula/geant4.rb:80]: Was python enabled by default before? If not, we should keep multithreading for this formula and then have the new formula enable python

Comment (cxwx) [reply Formula/geant4.rb:80]: OK, thanks for advice

**Pull request closed without merging**